pkg/services/alerting/conditions/reducer_test.go

```diff
@@ -128,25 +128,124 @@ func TestSimpleReducer(t *testing.T) {
 			So(reducer.Reduce(series).Float64, ShouldEqual, float64(3))
 		})
 
-		Convey("diff one point", func() {
+		// diff function Test Suite
+		Convey("diff of one positive point", func() {
 			result := testReducer("diff", 30)
 			So(result, ShouldEqual, float64(0))
 		})
 
-		Convey("diff two points", func() {
+		Convey("diff of one negative point", func() {
+			result := testReducer("diff", -30)
+			So(result, ShouldEqual, float64(0))
+		})
+
+		Convey("diff of two positive points[1]", func() {
 			result := testReducer("diff", 30, 40)
 			So(result, ShouldEqual, float64(10))
 		})
 
-		Convey("diff three points", func() {
+		Convey("diff of two positive points[2]", func() {
+			result := testReducer("diff", 30, 20)
+			So(result, ShouldEqual, float64(-10))
+		})
+
+		Convey("diff of two negative points[1]", func() {
+			result := testReducer("diff", -30, -40)
+			So(result, ShouldEqual, float64(-10))
+		})
+
+		Convey("diff of two negative points[2]", func() {
+			result := testReducer("diff", -30, -10)
+			So(result, ShouldEqual, float64(20))
+		})
+
+		Convey("diff of one positive and one negative point", func() {
+			result := testReducer("diff", 30, -40)
+			So(result, ShouldEqual, float64(-70))
+		})
+
+		Convey("diff of one negative and one positive point", func() {
+			result := testReducer("diff", -30, 40)
+			So(result, ShouldEqual, float64(70))
+		})
+
+		Convey("diff of three positive points", func() {
 			result := testReducer("diff", 30, 40, 40)
 			So(result, ShouldEqual, float64(10))
 		})
 
+		Convey("diff of three negative points", func() {
+			result := testReducer("diff", -30, -40, -40)
```

Comment:
```suggestion
			result := testReducer("diff", -30, -40, -50)
```

Reply:
I think this makes it more clear that we look at the first and last value. It would be great if we could fix this for all tests with more then 2 values.